public/student_list.php

```diff
@@ -0,0 +1,106 @@
+<?php require_once(dirname(__FILE__)."/../modules/ensure_logged_in.php"); ?>
+<?php include "templates/header.php"; ?>
+
+<?php if(get_current_role() == 'admin' || get_current_role() == 'instructor') { ?>
+<?php
+
+require_once "../modules/models/user.php";
+require_once "../modules/models/course_section.php";
+require_once "../modules/models/course_section_student.php";
+require_once "../common.php";
+
+try {
+    $students = User::where(array('is_instructor' => '0', 'is_admin' => '0'));
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+try {
+    $course_sections = CourseSection::getAll();
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+try {
+    $course_sections_students = CourseSectionStudent::includes("user")->getAll();
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+
+?>
+
+<?php
+if ($students && count($students)) { ?>
+        <h2>All Students</h2>
+
+    <table>
+            <thead>
+                <tr>
+                    <th>#</th>
+                    <th>First Name</th>
+                    <th>Last Name</th>
+                    <th>Email Address</th>
+                    <th>Created At</th>
+                </tr>
+            </thead>
+            <tbody>
+        <?php foreach ($students as $row) { ?>
+            <tr>
+                <td><?php echo escape($row->id); ?></td>
+                <td><?php echo escape($row->first_name); ?></td>
+                <td><?php echo escape($row->last_name); ?></td>
+                <td><?php echo escape($row->email); ?></td>
+                <td><?php echo escape($row->created_at);  ?> </td>
+            </tr>
+        <?php } ?>
+        </tbody>
+    </table>
+    <?php } else { ?>
+        <blockquote>No Students found.</blockquote>
+    <?php }?>
+
+
+  <?php
+    if ($course_sections && count($course_sections)) { ?>
+        <h2>Section student list</h2>
+
+        <?php foreach($course_sections as $section) { ?>
+            
+            <h3>Section <?php echo $section->course_section_name ?> </h3>
+            <table>
+                <thead>
+                    <tr>
+                        <th>#</th>
+                        <th>First Name</th>
+                        <th>Last Name</th>
+                        <th>Email Address</th>
+                        <th>Created At</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <?php foreach($course_sections_students as $section_students) { 
+                        if ($section->id == $section_students->section_id) {
+                            $student = $section_students->user; ?>
```

Comment:
I just saw this 😅 hmm why is it done this way, from `CourseSection` you can access `CourseSectionStudent` and then the user. I don't think there a need to do it explicitly in this manner

Reply:
I'll be honest, I did it this way because its the way I could picture it in my head.

Reply:
sure, but I'll have you know that its not a good way of doing it nor is it a scalable way